### DISCUSSION TITLE
Lowering the Batch Size

### DISCUSSION THREAD

**Discussion by vb-vasudev** [jsforce/jsforce#1453] [Ideas]
I'm trying to lower the Batch Size. My understanding as per the Salesforce documentation is that this is defined at the Connection level. 

https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_changing_batch_size.htm

Can we configure that in the connection object's CallOptions? If not, is there any other way to do this?

**Comment by dicksonstephen** [comment]
Any update on this?

**Comment by paustint** [comment]
AFAIK this is not possible. JSForce uses the REST API and the docs linked in the ticket appear to only apply to the SOAP API

> The query() or queryMore() are SOAP methods and can’t be used with REST API calls.

**Comment by cristiand391** [comment]
moved this to GH discussions/ideas

The REST API supports this header:
https://developer.salesforce.com/docs/atlas.en-us.250.0.api_rest.meta/api_rest/headers_queryoptions.htm

jsforce doesn't expose it a way to set it yet.
